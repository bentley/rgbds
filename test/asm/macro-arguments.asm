mac: MACRO
	println "'mac \#':"
	for i, _NARG
		println strfmt("\\%d: <\1>", i+1)
		shift
	endr
	println
ENDM

	mac /* block
		...comment */ ; comment
	mac /*a*/ 1 , 2 /*b*/
	mac \
	c, d
	mac 1, 2 + /* another ;
		; comment */ 2, 3

	mac
	mac a,,
	mac ,,z
	mac a,,z
	mac ,a,b,c,
	mac ,,x,,
