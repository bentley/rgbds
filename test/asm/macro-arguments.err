warning: macro-arguments.asm(19): [-Wempty-macro-arg]
    Empty macro argument
warning: macro-arguments.asm(19): [-Wempty-macro-arg]
    Empty macro argument
warning: macro-arguments.asm(20): [-Wempty-macro-arg]
    Empty macro argument
warning: macro-arguments.asm(20): [-Wempty-macro-arg]
    Empty macro argument
warning: macro-arguments.asm(21): [-Wempty-macro-arg]
    Empty macro argument
warning: macro-arguments.asm(22): [-Wempty-macro-arg]
    Empty macro argument
warning: macro-arguments.asm(22): [-Wempty-macro-arg]
    Empty macro argument
warning: macro-arguments.asm(23): [-Wempty-macro-arg]
    Empty macro argument
warning: macro-arguments.asm(23): [-Wempty-macro-arg]
    Empty macro argument
warning: macro-arguments.asm(23): [-Wempty-macro-arg]
    Empty macro argument
warning: macro-arguments.asm(23): [-Wempty-macro-arg]
    Empty macro argument
