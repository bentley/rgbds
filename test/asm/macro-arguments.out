'mac ':

'mac  1,2':
\1: < 1>
\2: <2>

'mac 	c,d':
\1: <	c>
\2: <d>

'mac 1,2 +  2,3':
\1: <1>
\2: <2 +  2>
\3: <3>

